# --- T2-COPYRIGHT-NOTE-BEGIN ---
# T2 SDE: package/*/openrgb/hotfix-corsair-ddr5.patch
# Copyright (C) 2023 The T2 SDE Project
# 
# This Copyright note is generated by scripts/Create-CopyPatch,
# more information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License version 2 as used by the T2 SDE.
# --- T2-COPYRIGHT-NOTE-END ---

--- OpenRGB-release_0.8/Controllers/CorsairVengeanceController/CorsairVengeanceControllerDetect.cpp.vanilla	2023-05-11 16:21:53.906373364 +0200
+++ OpenRGB-release_0.8/Controllers/CorsairVengeanceController/CorsairVengeanceControllerDetect.cpp	2023-05-11 16:25:37.913389538 +0200
@@ -67,6 +67,17 @@
                     
                     ResourceManager::get()->RegisterRGBController(new_rgbcontroller);
                 }
+	    }
+
+	    for(unsigned char addr = 0x18; addr <= 0x1F; addr++)
+	    {
+	        if(TestForCorsairVengeanceController(busses[bus], addr))
+	        {
+		    CorsairVengeanceController*     new_controller    = new CorsairVengeanceController(busses[bus], addr);
+		    RGBController_CorsairVengeance* new_rgbcontroller = new RGBController_CorsairVengeance(new_controller);
+                    
+		    ResourceManager::get()->RegisterRGBController(new_rgbcontroller);
+		}
             }
         }
     }
